components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt

```diff
@@ -408,42 +412,53 @@ class DefaultAddonUpdater(
      * Handles notification actions related to addons that require additional permissions
      * to be updated.
      */
-    /** @suppress */
-    class NotificationHandlerService : IntentService("NotificationHandlerService") {
+    class NotificationHandlerService : Service() {
 
         private val logger = Logger("NotificationHandlerService")
 
         @VisibleForTesting
         internal var context: Context = this
 
-        public override fun onHandleIntent(intent: Intent?) {
-            if (intent == null) return
+        internal fun onHandleIntent(intent: Intent?) {
+            val addonId = intent?.getStringExtra(NOTIFICATION_EXTRA_ADDON_ID) ?: return
 
             when (intent.action) {
                 NOTIFICATION_ACTION_ALLOW -> {
-                    handleAllowAction(intent)
+                    handleAllowAction(addonId)
                 }
                 NOTIFICATION_ACTION_DENY -> {
-                    removeNotification()
+                    removeNotification(addonId)
                 }
             }
         }
 
         @VisibleForTesting
-        internal fun removeNotification() {
-            val notificationId =
-                SharedIdsHelper.getIdForTag(context, NOTIFICATION_TAG)
+        internal fun removeNotification(addonId: String) {
+            val notificationId = getNotificationId(context, addonId)
             NotificationManagerCompat.from(context).cancel(notificationId)
         }
 
         @VisibleForTesting
-        internal fun handleAllowAction(intent: Intent) {
-            val addonId = intent.getStringExtra(NOTIFICATION_EXTRA_ADDON_ID)!!
+        internal fun handleAllowAction(addonId: String) {
             val storage = UpdateStatusStorage()
             logger.info("Addon $addonId permissions were granted")
             storage.markAsAllowed(context, addonId)
             GlobalAddonDependencyProvider.requireAddonUpdater().update(addonId)
-            removeNotification()
+            removeNotification(addonId)
+        }
+
+        override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+            onHandleIntent(intent)
+            return super.onStartCommand(intent, flags, startId)
+        }
+
+        override fun onBind(intent: Intent?): IBinder? = null
+
+        companion object {
+            @VisibleForTesting
+            internal fun getNotificationId(context: Context, addonId: String): Int {
+                return SharedIdsHelper.getIdForTag(context, NOTIFICATION_TAG + addonId)
```

Comment:
nit: do you want this to be + "." + addonId. Separated by a dot instead? 
e.g. `mozilla.components.feature.addons.update.addonUpdater.12345`

Reply:
Sure 👍🏽